### PR TITLE
Update required dependencies for support dpcpp and mpi

### PR DESCRIPTION
**Pull request by PetrovKP**

### CODE REVIEW

File: daal4py/oneapi/__init__.py [hunk 1]
```diff
@@ -33,5 +33,20 @@
     os.environ['PATH'] = current_path + os.pathsep + os.environ['PATH']
     os.environ['PATH'] = installed_package_path + os.pathsep + os.environ['PATH']
 
-from _oneapi import *
-from _oneapi import _get_sycl_ctxt, _get_device_name_sycl_ctxt, _get_sycl_ctxt_params
+try:
+    from _oneapi import *
+    from _oneapi import _get_sycl_ctxt, _get_device_name_sycl_ctxt, _get_sycl_ctxt_params
+except ModuleNotFoundError:
+    raise
```
Comment (michael-smirnov) [inline daal4py/oneapi/__init__.py:40]: Why you need this?

File: daal4py/oneapi/__init__.py [hunk 2]
```diff
@@ -33,5 +33,20 @@
...
+except ImportError:
+    import daal4py
+    version = daal4py._get__version__()[1:-1].split(', ')
+    major_version, minnor_version = version[0], version[1]
```
Comment (michael-smirnov) [inline daal4py/oneapi/__init__.py:44]:
```suggestion
    major_version, minor_version = version[0], version[1]
```

File: daal4py/oneapi/__init__.py [hunk 3]
```diff
@@ -33,5 +33,20 @@
...
+    raise ImportError(
+        f'dpcpp_cpp_rt >= {major_version}.{minnor_version} '
```
Comment (michael-smirnov) [inline daal4py/oneapi/__init__.py:46]: Are you sure that versions of daal4py and dpcpp_cpp_rt are aligned?

File: README.md [hunk 4]
```diff
@@ -90,6 +90,23 @@ conda install scikit-learn-intelex -c intel
 
 </details>
 
+⚠️ Note: *dpcpp support is optional dependency (required dependencies for support dpcpp
```
Comment (michael-smirnov) [inline README.md:93]: There is no information why the user needs to install this dependency.

File: daal4py/oneapi/__init__.py (hunk 1 above)
Comment (PetrovKP) [reply daal4py/oneapi/__init__.py:40]: We use in examples this exception: https://github.com/intel/scikit-learn-intelex/blob/master/examples/run_examples.py#L61

For example, Mac Os don't build daal4py with dpcpp support, so daal4py.oneapi should throw ModuleNotFoundError

File: daal4py/oneapi/__init__.py (hunk 3 above)
Comment (PetrovKP) [reply daal4py/oneapi/__init__.py:46]: Yes

File: README.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ Running the latest scikit-learn test suite with Intel(R) Extension for Scikit-le
 
 ## 👀 Follow us on Medium
 
-We publish blogs on Medium, so [follow us](https://medium.com/intel-analytics-software/tagged/machine-learning) to learn tips and tricks for more efficient data analysis the help of Intel(R) Extension for Scikit-learn. Here are our latest blogs:
+We publish blogs on Medium, so [follow us](https://medium.com/intel-analytics-software/tagged/machine-learning) to learn with the help of Intel(R) Extension for Scikit-learn. Here are our latest blogs:
```
Comment (outoftardis) [inline README.md:17]: the sentence doesn't make much sense :)

learn what?

File: README.md [hunk 6]
```diff
@@ -90,6 +90,23 @@ conda install scikit-learn-intelex -c intel
 
 </details>
 
+⚠️ Note: *dpcpp support is optional dependency (required dependencies for support dpcpp
+will not be downloaded). Need to manually install ***dpcpp_cpp_rt*** package*
```
Comment (outoftardis) [inline README.md:94]:
```suggestion
⚠️ Note: *DPC++ support is an optional dependency. Required dependencies for DPC++ support
will not be downloaded. You need to manually install ***dpcpp_cpp_rt*** package.*
```

File: daal4py/README.md [hunk 7]
```diff
@@ -56,6 +56,32 @@ conda install daal4py -c conda-forge
 conda install daal4py -c intel
 ```
 
+⚠️ Note: *dpcpp and MPI support is optional dependency (required dependencies for dpcpp
+MPI support will not be downloaded). Need to manually install ***dpcpp_cpp_rt*** package
+for support dpcpp and ***impi_rt*** for support MPI.*
```
Comment (outoftardis) [inline daal4py/README.md:61]:
```suggestion
⚠️ Note: *DPC++ and MPI support are optional dependencies. Required dependencies for DPC++ and
MPI support will not be downloaded. You need to manually install ***dpcpp_cpp_rt*** package
for DPC++ support and ***impi_rt*** package for MPI support.*
```

File: daal4py/oneapi/__init__.py [hunk 8]
```diff
@@ -33,5 +33,20 @@
...
+        'needs to be installed or upgraded in order to use this module.\n'
```
Comment (outoftardis) [inline daal4py/oneapi/__init__.py:47]:
```suggestion
        'has to be installed or upgraded to use this module.\n'
```

File: daal4py/oneapi/__init__.py [hunk 9]
```diff
@@ -33,5 +33,20 @@
...
+        'You can download or upgrade with help command:\n'
```
Comment (outoftardis) [inline daal4py/oneapi/__init__.py:48]:
```suggestion
        'You can download or upgrade it using the following commands:\n'
```

File: README.md (hunk 5 above)
Comment (PetrovKP) [reply README.md]: applied

File: README.md (hunk 6 above)
Comment (PetrovKP) [reply README.md]: applied

File: daal4py/README.md (hunk 7 above)
Comment (PetrovKP) [reply daal4py/README.md]: applied

File: daal4py/oneapi/__init__.py (hunk 9 above)
Comment (PetrovKP) [reply daal4py/oneapi/__init__.py]: applied

File: README.md (hunk 4 above)
Comment (PetrovKP) [reply README.md]: that's better?

File: README.md [hunk 10]
```diff
@@ -90,8 +90,8 @@ conda install scikit-learn-intelex -c intel
 
 </details>
 
-⚠️ Note: *dpcpp support is optional dependency (required dependencies for support dpcpp
-will not be downloaded). Need to manually install ***dpcpp_cpp_rt*** package*
+⚠️ Note. *GPU support is an optional. Required dependencies for GPU support
```
Comment (outoftardis) [inline README.md:93]:
```suggestion
⚠️ Note: *GPU support is an optional dependency. Required dependencies for GPU support
```

File: daal4py/README.md [hunk 11]
```diff
@@ -56,9 +56,9 @@ conda install daal4py -c conda-forge
 conda install daal4py -c intel
 ```
 
-⚠️ Note: *dpcpp and MPI support is optional dependency (required dependencies for dpcpp
-MPI support will not be downloaded). Need to manually install ***dpcpp_cpp_rt*** package
-for support dpcpp and ***impi_rt*** for support MPI.*
+⚠️ Note. *GPU and MPI support are optional dependencies.
```
Comment (outoftardis) [inline daal4py/README.md:59]:
```suggestion
⚠️ Note: *GPU and MPI support are optional dependencies.
```

**Review by outoftardis — APPROVED** [review]
no other comments from me

**Review by michael-smirnov — APPROVED** [review]

**Pull request merged**